src/services/eventHandlers.ts

```
@@ -525,6 +525,83 @@ export class EventHandlers {
     this.orthographicCamera = camera;
   }
 
+  /**
+   * Connect to the event bus for decoupled communication with SceneManager.
+   * This replaces the circular dependency pattern.
+   */
+  public setEventBus(eventBus: SceneEventBus): void {
+    // Clean up previous subscriptions
+    this.eventBusUnsubscribers.forEach(unsub => unsub());
```

Biome (2.3.13)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)